services/lemmy/lemmy.service.js

```diff
@@ -0,0 +1,94 @@
+import Joi from 'joi'
+import { metric } from '../text-formatters.js'
+import { BaseJsonService, InvalidParameter } from '../index.js'
+
+const queryParamSchema = Joi.object({
+  server_fqdn: Joi.string().hostname(),
+}).required()
+
+const lemmyCommunitySchema = Joi.object({
+  community_view: Joi.object({
+    counts: Joi.object({
+      subscribers: Joi.number().required(),
+      posts: Joi.number().required(),
+      comments: Joi.number().required(),
+    }).required(),
+  }).required(),
+}).required()
+
+export default class Lemmy extends BaseJsonService {
+  static category = 'social'
+
+  static route = {
+    base: 'lemmy',
+    pattern: ':community',
+    queryParamSchema,
+  }
+
+  static examples = [
+    {
+      title: 'Lemmy',
+      namedParams: { community: 'asklemmy@lemmy.ml' },
+      staticPreview: this.render({
+        community: 'asklemmy@lemmy.ml',
+        members: 42,
+      }),
+    },
+    {
+      title: 'Lemmy',
+      namedParams: { community: 'asklemmy@lemmy.ml' },
+      queryParams: { server_fqdn: 'lemmy.world' },
+      staticPreview: this.render({
+        community: 'asklemmy@lemmy.ml',
+        members: 42,
+      }),
+    },
+  ]
+
+  static _cacheLength = 30
+
+  static defaultBadgeData = { label: 'community' }
+
+  static render({ community, members }) {
+    return {
+      label: `subscribe to ${community}`,
+      message: `${metric(members)} subscribers`,
+      color: 'brightgreen',
+    }
+  }
+
+  async fetch({ community, serverFQDN }) {
+    let host
+    if (serverFQDN === undefined) {
+      const splitAlias = community.split('@')
+      // The community will be in the format of `community@server`
+      if (splitAlias.length !== 2) {
+        throw new InvalidParameter({
+          prettyMessage: 'invalid community',
+        })
+      }
+
+      host = splitAlias[1]
+    } else {
+      host = serverFQDN
```

Comment:
Can you explain what's going on here with being able to extract the URL we're going to query from either a param or the community name?

Are there situations where the community name is `something@thisurl.com` but we want to query `thaturl.com` for the data?

Reply:
We can technically lookup a community on the community's instance, or use another instance to lookup a community.

This may be helpful for adding kbin communities for example.

Reply:
I've heard of lemmy, but I don't know what kbin is. I'm not familiar enough with it to connect the dots of why this would be more useful or necessary in that context.

Can you give an example?

Reply:
Kbin is another federated platform like lemmy, compatible with Lemmy. Users on lemmy and kbin can interact with them. Kbin has it's own API, but with this change, we can use Lemmy's API to view data from kbin.

Reply:
For example, see https://lemmy.ml/c/gnome@kbin.social

Reply:
OK. Got it. So

- https://lemmy.ml/api/v3/community?name=asklemmy%40lemmy.ml is fine (and tell me the same number of subscribers as https://lemmy.ml/c/asklemmy )
- https://kbin.social/api/v3/community?name=gnome%40kbin.social doesn't work

but as a workaround, you could do

- https://lemmy.ml/api/v3/community?name=gnome%40kbin.social or
- https://lemmy.world/api/v3/community?name=gnome%40kbin.social

A question about that:

- If I look at https://lemmy.ml/api/v3/community?name=gnome%40kbin.social it tells me `gnome@kbin.social` has got 2 subscribers.
- If I look at https://lemmy.world/api/v3/community?name=gnome%40kbin.social it says `gnome@kbin.social` has got 29 subscribers.
- ..but if I look at https://kbin.social/m/GNOME it says 245 subscribers.

Any idea why the counts are so wildly different? I guess each instance only knows about some subset of `gnome@kbin.social` it is federated with (maybe lemmy.ml only knows about subscribers to `gnome@kbin.social` whose home instance is lemmy.ml?) Given that constraint, is this actually a useful feature? i.e: If you look up the stats for a community on some other non-home instance, you're always going to be getting way less than the full picture.

Feels like this feature (looking up on a non-home instance) could be something that technically works, but may not provide a metric anyone actually wants to display.

Reply:
I think the counts are different since it likely shows the number of subscribers from that instance, which does make it a bit useless. Should I remove this feature then?

Reply:
I've just removed the ability to specify FQDNs to fetch from in the latest commit.

Reply:
OK. I think this makes sense. We can always revisit it as a feature if there is a use-case for it.